Tests/Entity/SnapshotManagerTest.php

```diff
@@ -18,6 +18,78 @@
  */
 class SnapshotManagerTest extends \PHPUnit_Framework_TestCase
 {
+    public function testSetTemplates()
+    {
+        $manager = $this->getMockBuilder('Sonata\PageBundle\Entity\SnapshotManager')
+            // this method is called in SonataPageExtension
+            ->setMethods(array(
+                'setTemplates'
```

Comment:
In this case it works:
```php
$manager = $this->getMockBuilder('Sonata\PageBundle\Entity\SnapshotManager')
    ->setMethods(array(
    'foo',
    ))
    ->disableOriginalConstructor()
    ->getMock()
;
```
In this case does not work,  because we mock the method:
```php
$manager = $this->getMockBuilder('Sonata\PageBundle\Entity\SnapshotManager')
    ->setMethods(array(
    'setTemplates',
    ))
    ->disableOriginalConstructor()
    ->getMock()
;
```
But why does not work in these two cases?
```php
$manager = $this->getMockBuilder('Sonata\PageBundle\Entity\SnapshotManager')
    ->disableOriginalConstructor()
    ->getMock()
;


$manager = $this->getMockBuilder('Sonata\PageBundle\Entity\SnapshotManager')
    ->setMethods(array())
    ->disableOriginalConstructor()
    ->getMock()
;
```

Why we need to call the `setMethods` method with any value?